app/views/govuk_publishing_components/components/_fieldset.html.erb

```diff
@@ -5,29 +5,28 @@
 
   text = text || yield
   describedby ||= nil
-  role ||= nil
   heading_level ||= nil
   heading_size = false unless shared_helper.valid_heading_size?(heading_size)
   error_message ||= nil
   error_id ||= nil
-  id ||= nil
 
   if error_message
     error_id = "error-#{SecureRandom.hex(4)}" unless error_id
     describedby = error_id
   end
 
   component_helper = GovukPublishingComponents::Presenters::ComponentWrapperHelper.new(local_assigns)
-  component_helper.add_class("gem-c-fieldset govuk-form-group")
-  component_helper.add_class("govuk-form-group--error") if error_message
+  component_helper.add_class("govuk-fieldset")
+  component_helper.add_aria_attribute({ describedby: describedby }) if describedby
 
-  fieldset_classes = %w(govuk-fieldset)
+  css_classes = %w(gem-c-fieldset govuk-form-group)
+  css_classes << "govuk-form-group--error" if error_message
 
   legend_classes = %w(govuk-fieldset__legend)
   legend_classes << "govuk-fieldset__legend--#{heading_size}" if heading_size
 %>
-<%= tag.div(**component_helper.all_attributes) do %>
-  <%= tag.fieldset class: fieldset_classes, aria: { describedby: describedby }, role: role, id: id do %>
+<%= tag.div class: css_classes do %>
+  <%= tag.fieldset(**component_helper.all_attributes) do %>
```

Comment:
I'm slightly concerned that using the wrapper helper on an inner element opens the door to inconsistency and the risk that things that the wrapper does could be overridden or counteracted by styles on the parent. I think I need to understand this more: why does the fieldset need an id, and if it does need one, could we instead add a new param for `id_fieldset ||= nil` instead? (admittedly a breaking change, but seems like a cleaner solution?)

Reply:
I've opened a PR with the breaking change here:
https://github.com/alphagov/govuk_publishing_components/pull/4454